### ISSUE TITLE
Make `ServerExtension.webClient()` trust the self-signed certification of the `Server` which the client belongs to

### ISSUE THREAD

**Issue by ikhoon** [line/armeria#4013]
`ClientFactory.insecure()` will be used by default for `ServerExtension.webClient()`.
https://github.com/line/armeria/pull/3890#discussion_r734362244
As a follow-up issue, it might be better only not to verify the self-signed certification 
of the `ServerExtension` which the `WebClient` belongs to.

**Comment by trustin** [comment]
Agreed. This could be a potential security issue when a user involves an external endpoint that's potentially compromised.
